MediaBrowser.Common/System/OperatingSystem.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using MediaBrowser.Model.System;
+
+namespace MediaBrowser.Common.System
+{
+    public static class OperatingSystem
+    {
+        // We can't use Interlocked.CompareExchange for enums
+        private static int _id = int.MaxValue;
+
+        public static OperatingSystemId Id
+        {
+            get
+            {
+                if (_id == int.MaxValue)
+                {
+                    Interlocked.CompareExchange(ref _id, (int)GetId(), int.MaxValue);
```

Comment:
Do you really need all this complexity? Declaring internal id as enum and doing simple thing like
```cs
if (_id == OperatingSystemId.Unknown) _id = GetId();
return _id;
```
would lead to internal `_id` field being rewritten at most by a couple of parallel threads which hit it before first thread finishes initializing it... but they all initialize it to the same value.

Thus all this atomic swapping dance is unnecessary (in this particular case) and just clutters the understanding albeit looking clever :)

Reply:
The standard lib uses the same for ‘Environment’, so...

Reply:
We're not writing a standard lib here, and not everyone who's maintaining this project has enough knowledge of how this works ;) but whatever.

Reply:
Perhaps just a comment explaining in detail the logic?

Reply:
The logic itself is clear (for me at least), but that's because I'm somewhat used to parallel programming.
A comment would need to explain how atomic swaps work, and why it's necessary, and so on... a bit lengthy.